src/core/middleware.ts

```diff
@@ -135,7 +136,7 @@ export function hasNewModel<T>(
         req.model = await getRepository(entity).findOne({
           where: {
             [prop]: req.params[prop as string]
-          },
+          } as any,
```

Comment:
Why not `T` instead of `any` and extend the `Express.Request` interface with the `model:? T` property in the method  returned by  `wrapAsync`?

Reply:
I've updated the cast to `as T` but unfortunately there's no easy way to type `req.model` properly as it's passed around in a lot of places internally in express. As this is legacy code I'm going to leave it as is.

Reply:
You could try to define this property using "Declaration Merging" like this:

```ts
namespace global {
  namespace Express {
    interface Request {
      model?: T;
    }
  }
}
```

But I am not sure how we can define the Type Parameter there.

Reply:
sorry, my example was wrong, I have fixed it by replace `namespace global` with `declare global`

Reply:
If this is too complicated, please mark it as solved

Reply:
Ah we already do define it on Request in `src/typings/index.d.ts`, but with type `unknown`. The problem is really the generic typing, it's not really possible without changes to express itself.